libs/cart/src/selectors/cart/cart.selector.spec.ts

```diff
@@ -323,9 +328,45 @@ describe('Cart | Selector | Cart', () => {
     });
   });
 
+  describe('selectHasBillingAddress', () => {
+    it('selects whether the cart has a billing address', () => {
+			const selector = store.pipe(select(selectHasBillingAddress));
+      const expected = cold('a', {a: !!cart.billing_address});
+
+      expect(selector).toBeObservable(expected);
+    });
+  });
+
+  describe('selectHasShippingAddress', () => {
+    it('selects whether the cart has a shipping address', () => {
+			const selector = store.pipe(select(selectHasShippingAddress));
+      const expected = cold('a', {a: !!cart.shipping_address});
+
+      expect(selector).toBeObservable(expected);
+    });
+  });
+
+  describe('selectHasShippingMethod', () => {
+    it('selects whether the cart has a selected shipping method', () => {
+			const selector = store.pipe(select(selectHasShippingMethod));
+      const expected = cold('a', {a: !!cart.shipping_information});
+
+      expect(selector).toBeObservable(expected);
+    });
+  });
+
+  describe('selectHasPaymentMethod', () => {
+    it('selects whether the cart has a selected payment method', () => {
+			const selector = store.pipe(select(selectHasPaymentMethod));
+      const expected = cold('a', {a: !!cart.payment});
+
+      expect(selector).toBeObservable(expected);
+    });
+  });
+
   describe('selectCanPlaceOrder', () => {
-    it('selects whether the order has all the required fields', () => {
-      const canPlaceOrder = cart.items.length > 0 && cart.billing_address && cart.shipping_address && cart.shipping_information && !!cart.payment;
+    it('selects whether the cart has all the required fields for placing an order', () => {
+      const canPlaceOrder = cart.items.length > 0 && !!cart.billing_address && !!cart.shipping_address && !!cart.shipping_information && !!cart.payment;
```

Comment:
You mean just hardcode `const canPlaceOrder = false;`? That would work for the current version of the stubbed order but if the other fields are filled out for some reason then this expectation would fail. I can change if you want though.

Reply:
I think the issue is moreso that the test breaches the boundaries of the internal functions. The test as written crosses multiple layers of functionality:

1. Is the stubbed order in a placeable state?
2, Are the inner functions (has*) written correctly

When reading this test, I would expect one of two things:
1. A stubbed order that should absolutely, and one where it should absolutely fail.
2. Mocks to the inner functions.

Reply:
I see what you mean. I'll isolate true and false scenarios for each of these specs.

Reply:
changed. LMK if you are happy with it.